### DISCUSSION TITLE
How to manager time windows for a specific vehicle at a specific node

### DISCUSSION THREAD

**Discussion by FreddyHunter** [google/or-tools#2750] [Routing (and legacy CP) questions] [answered]
Hi Community,

Imagine the following situation: a specific location works from 8am to 5pm, and can be visited by some vehicles at any time during this time windows, but for others vehicles can only be visited at a restricted time windows (for example, from 10am to 1pm). How can I model it?

I already set the time windows for each location and I already set the time windows for each vehicle. Now, I need to know how to set time windows for a location based on a specific vehicle. Any suggestion?

Thanks in advance.

**Comment by Mizux** [comment, marked as answer]
Did you try to:
1. duplicate the location so you can model each time windows
2. Restrict the vehicle list of each location using `routing.VehicleVar()` 
3. Add a disjunction so solver only have to visit one of these nodes

EDIT:
here is an example: https://github.com/google/or-tools/blob/master/ortools/constraint_solver/samples/vrp_time_windows_per_vehicles.py

possible output:
```sh
./vrp_time_windows_per_vehicles.py
Objective: 467
Dropped nodes: 2 6 7 9 10 11 12 13 14 15 16 17(1) 19(3) 20(4) 21(5) 23(7) 24(8) 25(9) 27(11) 28(12) 29(13) 31(15) 33(1) 34(2) 35(3) 36(4) 37(5) 38(6) 40(8) 41(9) 42(10) 46(14) 48(16) 49(1) 50(2) 51(3) 52(4) 53(5) 54(6) 55(7) 56(8) 58(10) 59(11) 60(12) 61(13) 62(14) 63(15) 64(16)
Route for vehicle 0:
0 Time:537 -> 8 Time:540 -> 5 Time:567 -> 4 Time:598 -> 3 Time:624 -> 1 Time:652 -> 0 Time:683
Duration of the route:146min

Route for vehicle 1:
0 Time:655 -> 30(14) Time:660 -> 32(16) Time:687 -> 26(10) Time:716 -> 18(2) Time:745 -> 22(6) Time:773 -> 0 Time:804
Duration of the route:149min

Route for vehicle 2:
0 Time:776 -> 44(12) Time:780 -> 43(11) Time:806 -> 47(15) Time:834 -> 45(13) Time:864 -> 39(7) Time:892 -> 0 Time:919
Duration of the route:143min

Route for vehicle 3:
0 Time:898 -> 57(9) Time:900 -> 0 Time:927
Duration of the route:29min

Total duration of all routes: 467min
```

**Comment by FreddyHunter** [comment]
Hi Mizux,

Thanks for your reply, what I need is: the location operates from 8am to
5pm, a specific vehicle also operates from 8am to 5pm, but for this vehicle
visit this specific location is just allowed from 11am to 1pm as an example.

So, I did not understand why to model it with your suggestions. Can you
clarify it? Can you share something with me?

What I did right now is to set the time windows for each location. I also
set the time windows for each vehicle start and end. What I would like to
know is how to model if a vehicle is visiting a specific node set a
restricted time windows for this node.
